src/ItspServices.pServer.Authorization/Policy/UserDataOwnerCheck.cs

```diff
@@ -0,0 +1,21 @@
+﻿using ItspServices.pServer.Authorization.Authorizer;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ItspServices.pServer.Authorization.Policy
+{
+    public class UserDataOwnerCheck : UserDataAuthorizerDecorator
+    {
+        public UserDataOwnerCheck(IUserDataAuthorizer authorizer)
+            : base(authorizer)
+        {
+        }
+
+        public override bool Authorize()
+        {
+            bool isOwner = User.Id == Data.OwnerId;
+            return isOwner || _userDataAuthorizer.Authorize();
```

Comment:
This would be then simpler, when you explicitly implement the `Authorize()` method in the base decorator.